core/src/code.rs

```diff
@@ -515,6 +545,7 @@ impl Code {
             TryNewCodeConfig {
                 version,
                 stack_height,
+                export_stack_height: true,
```

Comment:
What the reason to set it as true here. Also cannot see reasons to make logic of this export creating in core. You can use wasm-instruments in your crate `calc-stack-height` in order to insert this export after creating new code:
```rust
let code = Code::try_new_mock_with_rules(
        inf_recursion.as_ref().to_vec(),
        |module| schedule.rules(module),
        TryNewCodeConfig {
            version: schedule.instruction_weights.version,
            stack_height: Some(u32::MAX),
            export_stack_height: true,
            ..Default::default()
        },
    )
    .map_err(|e| anyhow::anyhow!("{e}"))?;
    *code.code_mut() = insert_stack_height_export(code.code())
```

Reply:
> What the reason to set it as true here? 

Artifact

> Also cannot see reasons to make logic of this export creating in gear-core

We cannot guarantee global index is persistent

Reply:
> We cannot guarantee global index is persistent

ok